### PR TITLE
Feature: links for mileages

### PR DESCRIPTION
**Pull request by kamilsdz**

### CODE REVIEW

File: app/controllers/mileages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class MileagesController < ApplicationController
+  def index
+    @mileages = current_user.mileages.order(:car_id, meter_status: :asc).page params[:page]
```
Comment (fab-ian) [inline app/controllers/mileages_controller.rb]: Please change the sort order to ``` created_at: :desc ```

File: app/controllers/mileages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  def show
+    @mileage = current_user.mileages.find(params[:id])
```
Comment (fab-ian) [inline app/controllers/mileages_controller.rb]: You don't need using 'current_user' to fetch the wanted Mileage. Please use ``` Mileage.find(params[:id]) ```

I know this solution was earlier used in other 'controllers' but I didn't notice it before.

File: app/views/mileages/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.autopagerize_page_element
+  %table.table.table-hover
+    %thead
+      %tr
+        %th{:scope => "col"} #{t('.car')}
```
Comment (fab-ian) [inline app/views/mileages/index.html.haml]: I think this is more readable.
```  %th{scope: 'col'}= t('.car') ```
compared
```  %th{:scope => "col"} #{t('.car')} ```

File: app/views/mileages/index.html.haml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        %th{:scope => "col"} #{t('.meter_status')}
+        %th{:scope => "col"} #{t('.date')}
+    %tbody
+      - @mileages.each do |mileage|
+        %tr
+          %td #{ mileage.car.decorate.name }
```
Comment (fab-ian) [inline app/views/mileages/index.html.haml]: Similar example
``` %td= mileage.car.decorate.name ```

File: config/locales/pl.yml [hunk 5]
```diff
@@ -1,6 +1,17 @@
 pl:
   auth_admin_alert: 'Strona dostępna tylko dla administratorów'
 
+  mileages:
```
Comment (fab-ian) [inline config/locales/pl.yml]: We dont' need this translation.
```
4.5 Translations for Active Record Models
You can use the methods Model.model_name.human and Model.human_attribute_name(attribute) to transparently look up translations for your model and attribute names.
```
http://guides.rubyonrails.org/i18n.html#translations-for-active-record-models

File: app/controllers/mileages_controller.rb (hunk 2 above)
Comment (nikos83) [reply app/controllers/mileages_controller.rb]: Do we need restrict user mileages to be visible only for current user?

Comment (fab-ian) [reply app/controllers/mileages_controller.rb]: @nikos83  It's a 'show' action which returns only ONE specific record defined by his ID.

Comment (nikos83) [reply app/controllers/mileages_controller.rb]: I know but at the moment on the top url you can change id

Comment (fab-ian) [reply app/controllers/mileages_controller.rb]: @nikos83 And what's the problem?

Comment (kamilsdz) [reply app/controllers/mileages_controller.rb]: I think that @nikos83 is right. User should have access only for his entries. So, we just can add rescue for this (at this moment we have query error:
`ActiveRecord::RecordNotFound in MileagesController#show
Couldn't find Mileage with 'id'=7 [WHERE "mileages"."deleted_at" IS NULL AND "cars"."deleted_at" IS NULL AND "cars"."user_id" = $1]`

Comment (fab-ian) [reply app/controllers/mileages_controller.rb]: ok :)

File: app/controllers/mileages_controller.rb [hunk 6]
```diff
@@ -2,10 +2,13 @@
 
 class MileagesController < ApplicationController
   def index
-    @mileages = current_user.mileages.order(:car_id, meter_status: :asc).page params[:page]
+    @mileages = current_user.mileages.order(:car_id, created_at: :desc).page params[:page]
   end
 
   def show
     @mileage = current_user.mileages.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
```
Comment (fab-ian) [inline app/controllers/mileages_controller.rb]: Please remove this. It's a bad way for handling 404 errors. You don't get page status 404 what is useful in tests.
```
Started GET "/mileages/388888" for 127.0.0.1 at 2018-01-09 21:31:28 +0100
Processing by MileagesController#show as HTML
  Parameters: {"id"=>"388888"}
  User Load (0.9ms)  SELECT  "users".* FROM "users" WHERE "users"."deleted_at" IS NULL AND "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 4], ["LIMIT", 1]]
  Mileage Load (0.9ms)  SELECT  "mileages".* FROM "mileages" INNER JOIN "cars" ON "mileages"."car_id" = "cars"."id" WHERE "mileages"."deleted_at" IS NULL AND "cars"."deleted_at" IS NULL AND "cars"."user_id" = $1 AND "mileages"."id" = $2 LIMIT $3  [["user_id", 4], ["id", 388888], ["LIMIT", 1]]
Redirected to http://localhost:3000/mileages
Completed 302 Found in 14ms (ActiveRecord: 1.9ms
```
In your case it's difficult to manage 404 errors. 
To be consequent you should this use in ever 'edit/show/delete' action.
You have ready 404 pages in production which can be customize.

File: app/controllers/mileages_controller.rb [hunk 7]
```diff
@@ -2,10 +2,13 @@
 
 class MileagesController < ApplicationController
   def index
-    @mileages = current_user.mileages.order(:car_id, meter_status: :asc).page params[:page]
+    @mileages = current_user.mileages.order(:car_id, created_at: :desc).page params[:page]
```
Comment (fab-ian) [inline app/controllers/mileages_controller.rb]: Why do you sort by car_id?

**Comment by kamilsdz** [comment]
@fab-ian  
Probably you have right. It's better to use 404 page than rescue. :) I will fix it tomorow. And I don't sort by car_id - I use SQL Group by statement. I think that it's more clearly when list of mileages look like this:
* Opel Vectra 20000km
* Opel Vectra 21000km
*Opel Vectra 22000km
*Ford Focus 12000km
*Ford Focus 13000km

than when we don't group entries and they are sorted by created_at, for example:
* Opel Vectra 20000km
*Ford Focus 12000km
* Opel Vectra 21000km
*Ford Focus 13000km
*Opel Vectra 22000km

Do you think that it's okay?

**Comment by fab-ian** [comment]
@kamilsdz I think it's important to know which mileages are lately added, to prevent adding the same record again. 

In your proposal it's not easy to find the latest 'mileage' if you have a lot of records and for example three cars.

**Comment by kamilsdz** [comment]
Ok, I just change index method in MileagesController to:
`@mileages = current_user.mileages.page params[:page]`
It will be sorted by: id asc (= created_at asc)

**Pull request merged**